client/blocks/announcement-modal/docs/example.jsx

```diff
@@ -1,20 +1,55 @@
+import { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PlaceholderImage from 'calypso/assets/images/marketplace/plugins-revamp.png';
 import AnnouncementModal from 'calypso/blocks/announcement-modal';
-import PlaceholderImage from 'calypso/images/extensions/woocommerce/image-placeholder.png';
+import Button from 'calypso/components/forms/form-button';
+import { getCurrentUserId } from 'calypso/state/current-user/selectors';
+import { savePreference } from 'calypso/state/preferences/actions';
+import { hasReceivedRemotePreferences } from 'calypso/state/preferences/selectors';
+
+const AnnouncementModalExample = () => {
+	const announcementId = 'announcement-modal-example';
+	const [ show, setShow ] = useState( false );
+	const dispatch = useDispatch();
+	const userId = useSelector( ( state ) => getCurrentUserId( state ) );
+	const hasPreferences = useSelector( hasReceivedRemotePreferences );
+	const dismissPreference = `announcement-modal-${ announcementId }-${ userId }`;
+
+	if ( ! hasPreferences ) {
+		return null;
+	}
 
-export default function AnnouncementModalExample() {
 	const pages = [
 		{
 			heading: 'All the plugins and more',
 			content:
 				'This page may look different as we’ve made some changes to improve the experience for you. Stay tuned for even more exciting updates to come!',
 			featureImage: PlaceholderImage,
 		},
-		{
-			heading: 'All the plugins and more Page 2',
-			content:
-				'This page may look different as we’ve made some changes to improve the experience for you. Stay tuned for even more exciting updates to come!',
-			featureImage: PlaceholderImage,
-		},
 	];
-	return <AnnouncementModal announcementId="my-new-super-duper-feature" pages={ pages } />;
-}
+
+	const TriggerButton = () => (
+		<Button onClick={ () => dispatch( savePreference( dismissPreference, 0 ) ) && setShow( true ) }>
+			Show
+		</Button>
+	);
+
+	if ( show ) {
+		return (
+			<>
+				{ TriggerButton() }
```

Comment:
Why are we calling this as a function instead of rendering it as a component?
```suggestion
				<TriggerButton />
```

Reply:
Absolutely no reason, thanks for catching this.